protocols/luci-proto-batman-adv/htdocs/luci-static/resources/protocol/batadv.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+'require form';
+'require network';
+
+
+network.registerPatternVirtual(/^bat\d+/);
+
+return network.registerProtocol('batadv', {
+	getI18n: function() {
+		return _('B.A.T.M.A.N. advanced');
+	},
+
+	getIfname: function() {
+		return this._ubus('l3_device') || this.sid;
+	},
+
+	getOpkgPackage: function() {
+		return 'batman-adv';
+	},
+
+	isFloating: function() {
+		return true;
+	},
+
+	isVirtual: function() {
+		return true;
+	},
+
+	getDevices: function() {
+		return null;
+	},
+
+	containsDevice: function(ifname) {
+		return (network.getIfnameOf(ifname) == this.getIfname());
+	},
+
+	renderFormOptions: function(s) {
+		var dev = this.getL3Device() || this.getDevice(), o;
+
+
+		try {
+			s.tab('mesh', _('Mesh Routing'), _('Mesh and routing related options'));
+		}
+		catch(e) {}
+
+
+
+// @FIXME - the list of routing protocols should not be hard coded but come from batctl  
+
+		o = s.taboption('mesh', form.ListValue, 'routing_algo', _('Routing Algorithm'),_('The algorithm that is used to discover mesh routes'));
+		o.value('BATMAN_IV', 'BATMAN_IV');
+		o.value('BATMAN_V', 'BATMAN_V');
```

Comment:
Value Translation?
Maybe we can not use the abstract dropdown value here but a more expressive one with translation?

Reply:
These terms are fixed - I don't think translation makes sense. I leave this open